### PR TITLE
Bare metal boot stuff, VGA driver - review comments incorporated

### PR DESCRIPTION
**Pull request by phanikishoreg**

### CODE REVIEW

File: src/kernel/include/shared/cos_config.h [hunk 1]
```diff
@@ -119,4 +119,11 @@
 /* print out to dmesg? */
 /* #define COS_PRINT_DMESG 1 */
 
+/**
+ * Configuration to enable/disable functionality in Kernel.
+ */
+#define ENABLE_VGA
+#define ENABLE_SERIAL
+#define ENABLE_TIMER
```
Comment (gparmer) [inline src/kernel/include/shared/cos_config.h]: This commit conflicts with Jakobs.  Fix the merge conflict (sorry!).  I believe he removed `ENABLE_TIMER`.

File: src/platform/i386/Makefile [hunk 2]
```diff
@@ -41,7 +41,8 @@ OBJS += serial.o
 OBJS += timer.o
 OBJS += chal.o
 OBJS += boot_comp.o
-OBJS += console.o
+#OBJS += console.o
```
Comment (gparmer) [inline src/platform/i386/Makefile]: If this is commented out, isn't it going to do nothing when `ENABLE_SERIAL` is defined?  Why is this not compiled now?

File: src/platform/i386/isr.h [hunk 3]
```diff
@@ -6,11 +6,13 @@
 #include "chal_asm_inc.h"
 
 enum {
-	IRQ_PAGE_FAULT = 14,
-	IRQ_PIT        = 32,
-	IRQ_KEYBOARD   = 33,
-	IRQ_SERIAL     = 36
+	IRQ_DOUBLE_FAULT = 8,
+	IRQ_PAGE_FAULT   = 14,
+	IRQ_PIT          = 32,
```
Comment (gparmer) [inline src/platform/i386/isr.h]: You'll find that Jakob had to alter this as well.  Please make sure your changes are consistent with his.

File: src/platform/i386/kernel.c [hunk 4]
```diff
@@ -138,6 +147,9 @@ kmain(struct multiboot *mboot, u32_t mboot_magic, u32_t esp)
        	comp_init();
```
Comment (gparmer) [inline src/platform/i386/kernel.c]: Wow, why is the spacing off on these?

File: src/platform/i386/vga.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+	size_t i = 0;
+
+	cll();
+	for (i = 0; i < strnlen((const char*)text, STRLEN_MAX); i++) {
+		putchar(text[i]);
+	}
+	move_csr();
+}
+
+void keyboard_handler(struct pt_regs *regs)
```
Comment (gparmer) [inline src/platform/i386/vga.c]: Don't we already have a keyboard handler?  See Jakob's code.

File: src/kernel/include/shared/cos_config.h (hunk 1 above)
Comment (phanikishoreg) [inline src/kernel/include/shared/cos_config.h]: Will do that. :)

File: src/platform/i386/Makefile (hunk 2 above)
Comment (phanikishoreg) [inline src/platform/i386/Makefile]: Console.o is also a VGA driver but I had created a new one. I can even remove it totally. What do you think? ENABLE_SERIAL is for serial.o and not this one.

File: src/platform/i386/isr.h (hunk 3 above)
Comment (phanikishoreg) [inline src/platform/i386/isr.h]: Will check and resolve conflicts.

File: src/platform/i386/kernel.c (hunk 4 above)
Comment (phanikishoreg) [inline src/platform/i386/kernel.c]: I think it has been like that from before. I've not touched that part, there is no diff of base to my version at these lines. I'll fix it anyway in my next pull request. :)

File: src/platform/i386/vga.c (hunk 5 above)
Comment (phanikishoreg) [inline src/platform/i386/vga.c]: Do we have one? I'll check his code. This was the only thing I merged from console.o in this new vga.o.

**Comment by phanikishoreg** [comment]
Unmerged again!! :( 2 failed attempts in a row for me! :P

File: src/platform/i386/vga.c (hunk 5 above)
Comment (phanikishoreg) [inline src/platform/i386/vga.c]: Just checked, only keyboard_handler() is in console.o and that is not disabled in Jacob's code but in my commit, console.o is disabled and replaced with vga.o and I've included this handler in vga.o, that should be fine right?

**Pull request closed without merging**